### PR TITLE
[T6A1][W15-C3]

### PR DESCRIPTION
**Pull request by marcusngwj**

### CODE REVIEW

File: doc/UserGuide.md [hunk 1]
```diff
@@ -48,7 +48,18 @@ Examples:
 * `find John`<br>
   Returns `John Doe` but not `john`
 * `find Betsy Tim John`<br>
-  Returns Any person having names `Betsy`, `Tim`, or `John`
+  Returns Any person having names `Betsy`, `Tim`, or `John`  
+  
+## Finding all persons containing a keyword in their tags: `find-tag`
```
Comment (okkhoy) [inline doc/UserGuide.md]: is it `find` or `find-tag`? seems confusing

File: doc/UserGuide.md [hunk 2]
```diff
@@ -64,7 +75,46 @@ Examples:
...
+## Editing name of a person : `edit-name`
+Edits the name of the specified person in the address book. Irreversible.<br>
+Format: `edit-name INDEX n/NEW NAME`
+
+> Edits the name of the person at the specified `INDEX`. 
+  The index refers to the index number shown in the most recent listing.
+  
+Examples: 
+* `list`<br>
+  `edit 2 n/Jane`<br>
```
Comment (okkhoy) [inline doc/UserGuide.md]: again, `edit` or `edit-name`? if it is `edit-name` why `n/` again?

File: doc/UserGuide.md [hunk 3]
```diff
@@ -64,7 +75,46 @@ Examples:
...
+## Editing phone number of a person : `edit-phone`
+Edits the name of the specified person in the address book. Irreversible.<br>
+Format: `edit-phone INDEX n/NEW PHONENO`
+
+> Edits the phone number of the person at the specified `INDEX`. 
+  The index refers to the index number shown in the most recent listing.
+  
+Examples: 
+* `list`<br>
+  `edit 2 n/Jane`<br>
```
Comment (okkhoy) [inline doc/UserGuide.md]: bad copy paste alert here

File: doc/UserGuide.md [hunk 4]
```diff
@@ -64,7 +75,46 @@ Examples:
...
+## Editing email of a person : `edit-email`
+Edits the email of the specified person in the address book. Irreversible.<br>
+Format: `edit-email INDEX n/NEW EMAIL`
+
+> Edits the email of the person at the specified `INDEX`. 
+  The index refers to the index number shown in the most recent listing.
+  
+Examples: 
+* `list`<br>
+  `edit-email 2 n/Jane@gmail.com`<br>
```
Comment (okkhoy) [inline doc/UserGuide.md]: why `n/` if it is email?

File: src/seedu/addressbook/commands/EditPhoneCommand.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
+package seedu.addressbook.commands;
+
+import seedu.addressbook.common.Messages;
+import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.Person;
+import seedu.addressbook.data.person.Phone;
+import seedu.addressbook.data.person.UniquePersonList.PersonNotFoundException;
+
+/**
+ * Edit the name of a person in the address book.
```
Comment (okkhoy) [inline src/seedu/addressbook/commands/EditPhoneCommand.java]: uh! oh!! you need to be careful about these when you code... the  class is edit phone, but your comment is edit person name?

File: src/seedu/addressbook/commands/EditPhoneCommand.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+ */
+public class EditPhoneCommand extends Command{
+
+    public static final String COMMAND_WORD = "edit-phone";
+    
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ":\n" 
+            + "Edit the phone number of a person identified by the index number used.\n\t"
+            + "Parameters: INDEX p/NEW NUMBER\n\t"
```
Comment (okkhoy) [inline src/seedu/addressbook/commands/EditPhoneCommand.java]: this seems more appropriate than in the user guide; user guide says `n/` :-1:

File: src/seedu/addressbook/commands/FindByTagCommand.java [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.data.tag.Tag;
+import seedu.addressbook.data.tag.UniqueTagList;
+
+public class FindByTagCommand extends Command {
```
Comment (okkhoy) [inline src/seedu/addressbook/commands/FindByTagCommand.java]: helps to have some header comments here!

File: src/seedu/addressbook/data/person/Phone.java [hunk 8]
```diff
@@ -28,11 +28,33 @@ public Phone(String phone, boolean isPrivate) throws IllegalValueException {
         }
         this.value = phone;
     }
+    /**
+     * Store new valid phone number
+     */
+    public void storeNum(String newValue){
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Phone.java]: where is this used?

File: src/seedu/addressbook/parser/Parser.java [hunk 9]
```diff
@@ -227,6 +320,23 @@ private Command prepareFind(String args) {
         final Set<String> keywordSet = new HashSet<>(Arrays.asList(keywords));
         return new FindCommand(keywordSet);
     }
+    
+    /**
+     * Parses arguments in the context of the find by tags command.
```
Comment (okkhoy) [inline src/seedu/addressbook/parser/Parser.java]: these set of comments are apt :+1:

File: test/java/seedu/addressbook/commands/EditNameCommandTest.java [hunk 10]
```diff
@@ -0,0 +1,35 @@
+package seedu.addressbook.commands;
```
Comment (okkhoy) [inline test/java/seedu/addressbook/commands/EditNameCommandTest.java]: :+1:  for at least attempting this one!

**Comment by okkhoy** [comment]
@marcusngwj @turtle96 @Mamtha3005 @ZuoZhuolin 
Some comments added. Please ack & close the PR after reading comments.

**Comment by marcusngwj** [comment]
Noted. With thanks.

**Pull request closed without merging**